Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,342 @@
+      closestPoint[1] = raycastResult.closeY;
+    }
+  }
+  if (closestPoint[0] && closestPoint[1]) return closestPoint;
+  return null;
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.getRendererObject = function () {
+  // Mandatory, return the internal PIXI object used for your object:
+  if (this._debugMode) {
+    return this._debugLight;
+  }
+  return this._light;
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function () {
+  if (this._debugMode) this.updateGraphics();
+  this.updateBuffers();
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateGraphics = function () {
+  var raycastResult = this.raycastTest();
+  var vertices = new Array(2 * raycastResult.length + 2);
+  vertices[0] = this._object.x;
+  vertices[1] = this._object.y;
+
+  for (var i = 2; i < 2 * raycastResult.length + 2; i += 2) {
+    vertices[i] = raycastResult[i / 2 - 1][0];
+    vertices[i + 1] = raycastResult[i / 2 - 1][1];
+  }
+
+  this._graphics.clear();
+  for (var i = 0; i < vertices.length; i += 2) {
+    var lineColor = i % 4 === 0 ? 0xff0000 : 0x00ff00;
+    this._graphics
+      .lineStyle(1, lineColor, 1)
+      .moveTo(vertices[0], vertices[1])
+      .lineTo(vertices[i], vertices[i + 1]);
+    if (i !== vertices.length - 2) {
+      this._graphics.lineTo(vertices[i + 2], vertices[i + 3]);
+    } else {
+      this._graphics.lineTo(vertices[2], vertices[3]);
+    }
+  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateBuffers = function () {
+  this._center[0] = this._object.x;
+  this._center[1] = this._object.y;
+  this._light.shader.uniforms.center = this._center;
+
+  var raycastResult = this.raycastTest();
+  // Fallback to simple quad when there are no obstacles around.
+  if(raycastResult.length === 0) {
+    if(!this._defaultVertexBuffer) {
+      this._defaultVertexBuffer = new Float32Array(8);
+    }
+    if(!this._defaultIndexBuffer) {
+      this._defaultIndexBuffer = new Uint16Array([
+        0, 1, 2, 0, 2, 3
+      ]);
+    }
+
+    this._defaultVertexBuffer[0] = this._object.x - this._radius;
+    this._defaultVertexBuffer[1] = this._object.y + this._radius;
+    this._defaultVertexBuffer[2] = this._object.x + this._radius;
+    this._defaultVertexBuffer[3] = this._object.y + this._radius;
+    this._defaultVertexBuffer[4] = this._object.x + this._radius;
+    this._defaultVertexBuffer[5] = this._object.y - this._radius;
+    this._defaultVertexBuffer[6] = this._object.x - this._radius;
+    this._defaultVertexBuffer[7] = this._object.y - this._radius;
+
+    this._light.geometry
+      .getBuffer('aVertexPosition')
+      .update(this._defaultVertexBuffer);
+    this._light.geometry.getIndex().update(this._defaultIndexBuffer);
+    return;
+  }
+
+  var raycastResultLength = raycastResult.length;
+
+  var isSubArrayUsed = false;
+  var vertexBufferSubArray = null;
+  var indexBufferSubArray = null;
+
+  if (this._vertexBuffer.length > 2 * raycastResultLength + 2) {
```

Comment:
Let's add a comment describing what is this manipulation done on the subArray, explaining that it's an optimization to avoid re-allocating buffers :)